hypershift-operator/controllers/nodepool/nodepool_controller_test.go

```diff
@@ -1013,3 +1014,23 @@ func TestListMachineTemplatesIBMCloud(t *testing.T) {
 	g.Expect(err).ToNot(HaveOccurred())
 	g.Expect(len(templates)).To(Equal(0))
 }
+
+func TestGetName(t *testing.T) {
+	g := NewWithT(t)
+
+	alphaNumeric := regexp.MustCompile(`^[a-zA-Z0-9_]*$`)
```

Comment:
```suggestion
	alphaNumeric := regexp.MustCompile(`^[a-zA-Z0-9]*$`)
```

Reply:
Also I am pretty sure uppercase letters are not allowed, although it is probably not possible to get them because that would require the hcluster or nodepool name to contain an upperscore letter which isn't possible?

Reply:
nice catch, updated.